src/Config/ConfigFileInterface.php

```diff
@@ -0,0 +1,43 @@
+<?php namespace Winter\Storm\Config;
+
+// Returns are commented out as improved type variance is not supported until php 7.4
+// Ref: https://stitcher.io/blog/new-in-php-74#improved-type-variance-rfc
+// TODO: enable return types once support for <7.4 is dropped
+
+/**
+ * Interface ConfigFileInterface
+ * @package Winter\Storm\Config
+ */
+interface ConfigFileInterface
+{
+    /**
+     * Return a new instance of `ConfigFileInterface` ready for modification of the file.
+     *
+     * @param string $file
+     * @return ConfigFile|null
+     */
+    public static function read(string $file); //: ?ConfigFileInterface;
```

Comment:
```suggestion
    public static function open(string $filePath); //: ?ConfigFileInterface;
```
Matches the PHP core of `fopen()` `fwrite()` better